lisa/tools/chrony.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import re
+import time
+from typing import cast
+
+from retry import retry
+
+from lisa.executable import Tool
+from lisa.operating_system import Debian, Posix, Redhat, Suse
+from lisa.util import LisaException
+
+from .echo import Echo
+from .servicectl import Servicectl
+
+
+class Chrony(Tool):
+    # Leap status     : Normal
+    __leap_status_pattern = re.compile(
+        r"([\w\W]*?)Leap status.*:.*Normal$", re.MULTILINE
+    )
+    __no_server_set = "Number of sources = 0"
+
+    @property
+    def command(self) -> str:
+        return "chronyc"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def install(self) -> bool:
+        posix_os: Posix = cast(Posix, self.node.os)
+        posix_os.install_packages("chrony")
+        return self._check_exists()
+
+    def restart(self) -> None:
+        posix_os: Posix = cast(Posix, self.node.os)
+        if isinstance(self.node.os, Debian):
+            service_name = "chrony"
+        elif isinstance(self.node.os, Redhat) or isinstance(self.node.os, Suse):
+            service_name = "chronyd"
+        else:
+            raise LisaException(
+                f"Please double check chrony service name in {posix_os.release_version}"
+            )
+        servicectl = self.node.tools[Servicectl]
+        servicectl = servicectl.initialize_tool()
+        servicectl.restart_service(service_name)
+        time.sleep(5)
```

Comment:
Don't write hard code sleep, it's not stable, and may cost extra time. Use a condition to check it, and sleep 0.5 sec.